modules/gateway/conn.go

```diff
@@ -26,5 +27,10 @@ func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
 		Timeout: dialTimeout,
 		Cancel:  g.threads.StopChan(),
 	}
-	return dialer.Dial("tcp", string(addr))
+	conn, err := dialer.Dial("tcp", string(addr))
+	if err != nil {
+		return nil, err
+	}
+	conn.SetDeadline(time.Now().Add(connStdDeadline))
```

Comment:
Oops, I missed the `conn.SetDeadline(time.Time{})` added in another commit to reset the deadline in `Connect`.